urbanopt/geojson-modelica-translator
# refactor: add sys params to load partial templates

#### What does this PR accomplish?
- Adds building sys params to:
  - load model instance partial templates
  - component definitions partial templates (if coupling includes a load)
  - connect statements partial templates (if coupling includes a load)

Also includes changes from #419 which was merged into this PR's branch. TLDR is we moved some sys params around so that we can also drop them into other couplings.

Also includes changes from #418 which was merged into this PR's branch. TLDR we used the sys params from the district_system in the network 2 pipe <> Heating Plant coupling file

#### How should this be manually tested?
- Edit one of the templates to use `sys_params`, run a test which uses the template, verify it was templated into the resulting model.

#### What are the relevant tickets?
First step into solving #401 

#### Screenshots (if appropriate)
Example templating into a coupling file (⚠️ building sys params have moved since this screenshot, they're now at `sys_params.building`)
<img width="465" alt="Screen Shot 2021-08-04 at 10 29 22 AM" src="https://user-images.githubusercontent.com/18518728/128203710-de72f8fe-b548-430d-a7b2-ae7484e347db.png">

<img width="470" alt="Screen Shot 2021-08-04 at 10 28 56 AM" src="https://user-images.githubusercontent.com/18518728/128203786-aa1ef5ca-7454-4387-87cc-9c67d9ee8180.png">

Example templating into an instance file (⚠️ building sys params have moved since this screenshot, they're now at `sys_params.building`)
<img width="367" alt="Screen Shot 2021-08-04 at 10 30 10 AM" src="https://user-images.githubusercontent.com/18518728/128203827-729d70f0-3b78-4e94-912a-7731e7f51b09.png">

<img width="427" alt="Screen Shot 2021-08-04 at 10 29 47 AM" src="https://user-images.githubusercontent.com/18518728/128203835-382cc1c1-ff52-4c70-bca0-6d0bccd19cbd.png">


Example templating with district_system params
<img width="765" alt="Screen Shot 2021-08-13 at 4 47 56 PM" src="https://user-images.githubusercontent.com/18518728/129417459-7558e67f-c3e6-4055-b8a2-2d5674479693.png">

<img width="616" alt="Screen Shot 2021-08-13 at 4 47 38 PM" src="https://user-images.githubusercontent.com/18518728/129417472-71fe3ca3-a96b-423b-ada0-b78918c68659.png">
